### PR TITLE
General Update to ACHP

### PR DESCRIPTION
**Pull request by davideziviani**
Enhanced the structure of ACHP
Added components
Implemented AbstractState Class

### CODE REVIEW

File: ACHP/MultiCircuitEvaporator.py [hunk 1]
```diff
@@ -265,7 +260,7 @@ def Calculate(self):
         #Convert back to a single value for the overall evaporator
         self.Fins.Air.Vdot_ha=float(self.Fins.Air.Vdot_ha[-1])
         if self.Verbosity>0:
-            print(chr(127),end='') #progress bar
+            print (chr(127)) #progress bar
```
Comment (abahman) [inline ACHP/MultiCircuitEvaporator.py]: missing (,end='')

File: ACHP/MultiCircuitEvaporator.py [hunk 2]
```diff
@@ -296,94 +292,102 @@ def Calculate(self):
...
 
     csv_file= "Evaporator_MCE.csv"
-    print("\n demonstrating write to csv",csv_file,'\n')
-    Write2CSV(MCE,open(csv_file,'w'),append=False)
+    print ("\n demonstrating write to csv",csv_file,'\n')
+    #Write2CSV(MCE,open(csv_file,'w'),append=False)
       
-    print("check MCE capacity summation", MCE.hin_r*MCE.mdot_r[-1],end='')
-    print(np.sum([MCE.Evaps[i].hin_r*MCE.Evaps[i].mdot_r for i in range(MCE.Fins.Tubes.Ncircuits)]))
+    print ("check MCE capacity summation", MCE.hin_r*MCE.mdot_r[-1])
```
Comment (abahman) [inline ACHP/MultiCircuitEvaporator.py]: missing (,end='')

File: ACHP/VICompressorTello.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
+from __future__ import division, print_function, absolute_import
```
Comment (abahman) [inline ACHP/VICompressorTello.py]: @davideziviani 
Do you want to add VI comp cycle!?

Comment (davideziviani) [reply ACHP/VICompressorTello.py]: @abahman nope, not now. But this model is available in the public domain already.

File: ACHP/CoaxialHX.py [hunk 4]
```diff
@@ -1,14 +1,11 @@
 from __future__ import print_function, absolute_import
-
+from CoolProp.CoolProp import PropsSI
+from ACHP.Correlations import f_h_1phase_Annulus,f_h_1phase_Tube,ShahEvaporation_Average
```
Comment (ibell) [inline ACHP/CoaxialHX.py]: Inside the package, the imports should be .Correlations, not ACHP.Correlations (I think)

File: ACHP/CoaxialHX.py [hunk 5]
```diff
@@ -271,19 +274,27 @@ def _TwoPhase_Forward(self,w_2phase=1.0,xout_2phase=1.0):
...
-            print(Q_2phase_eNTU-self.Q_2phase)
+            print (Q_2phase_eNTU-self.Q_2phase)
         return Q_2phase_eNTU-self.Q_2phase
         
 if __name__=='__main__':
     
     TT=[]
     QQ=[]
     Q1=[]
+    #refigearnt Abstract State
```
Comment (ibell) [inline ACHP/CoaxialHX.py]: typo in refrigerant

File: ACHP/CoaxialHX.py [hunk 6]
```diff
@@ -292,15 +303,17 @@ def _TwoPhase_Forward(self,w_2phase=1.0,xout_2phase=1.0):
...
                 'hin_r':h,
-                'pin_r':PropsSI('P','T',Tdew_evap,'Q',1.0,'R290'),
+                'pin_r':PropsSI('P','T',Tdew_evap,'Q',1.0,Ref_r),
                 'pin_g':300000,     #pin_g in Pa
                 'Tin_g':290.52,
-                'Ref_r':'R290',
-                'Backend_r':'HEOS', #choose between: 'HEOS','TTSE&HEOS','BICUBIC&HEOS','REFPROP','SRK','PR'
-                'Ref_g':'Water',
-                'Backend_g':'HEOS', #choose between: 'HEOS','TTSE&HEOS','BICUBIC&HEOS','REFPROP','SRK','PR'
+                'AS_r':AS_r, #Abstract state of refigerant
```
Comment (ibell) [inline ACHP/CoaxialHX.py]: Inconsistent formatting of comments

File: ACHP/Condenser.py [hunk 7]
```diff
@@ -1,11 +1,9 @@
 from __future__ import division, print_function, absolute_import
 from math import pi,log,exp
-
-from .Correlations import f_h_1phase_Tube,ShahCondensation_Average,LMPressureGradientAvg,TwoPhaseDensity,AccelPressureDrop 
-from .FinCorrelations import WavyLouveredFins,FinInputs,IsFinsClass, HerringboneFins, PlainFins
-from .ACHPTools import ValidateFields
-
+from ACHP.Correlations import f_h_1phase_Tube,ShahCondensation_Average,LMPressureGradientAvg,TwoPhaseDensity,AccelPressureDrop
```
Comment (ibell) [inline ACHP/Condenser.py]: I think these too should be .Correlations, not ACHP.Correlations

File: ACHP/ExpDev.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
+from __future__ import division, absolute_import, print_function
+from CoolProp.CoolProp import PropsSI
+import CoolProp as CP
+
+from math import pi,exp,log,sqrt,tan,cos,sin
+
+class ExpDevClass():
+    """
+     Expansion device model
+     base on the work developed by Bo Shen in ACMODEL
```
Comment (ibell) [inline ACHP/ExpDev.py]: based

File: ACHP/OilPropLib.py [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+    return k_l
+
+    
+def Solubility_Ref_in_Liq(Ref,Liq,T,p):
+    x_Ref=0.0
+    Tmax=273.15
+    Tmin=273.15
+    error=False
+
+    if Ref=='R744' and Liq=='Water':
```
Comment (ibell) [inline ACHP/OilPropLib.py]: Reference?

File: ACHP/OilPropLib.py [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+        +8.76225186E-12*T**4-5.10876533E-05*p
+        +3.57823191E-09*p**2-1.37995983E-13*p**3
+        +2.05230067E-18*p**4+2.52713006E-07*T*p
+        -7.95947422E-12*T*p**2+8.30553293E-17*T*p**3
+        -6.90483618E-10*T**2*p+2.36780380E-14*T**2*p**2
+        -2.84627591E-19*T**2*p**3+6.06021155E-13*T**3*p
+        -2.13606178E-17*T**3*p**2+2.75290092E-22*T**3*p**3)
+        x_Ref=x_CO2
+    elif Ref=='R744' and Liq =='PAG':
+        Tmin=273.15+40
```
Comment (ibell) [inline ACHP/OilPropLib.py]: Reference?

File: ACHP/Preconditioners.py [hunk 11]
```diff
@@ -178,29 +184,30 @@ def OBJECTIVE(x):
...
             Cycle.Compressor.Tin_r=Tevap+Cycle.Evaporator.DT_sh
             Cycle.Compressor.Ref=Cycle.Ref
+            Cycle.Compressor.AS=Cycle.AS
             Cycle.Compressor.Calculate()
             W=Cycle.Compressor.W
             
             #Evaporator will be dry
             Qcoolingcoil=epsilon*Cycle.CoolingCoil.Fins.Air.Vdot_ha*rho_air*(Tin_CC-Cycle.CoolingCoil.Fins.Air.Tdb)*Cp_air
             
             #Glycol specifi heat
```
Comment (ibell) [inline ACHP/Preconditioners.py]: typo - specific

File: ACHP/convert_units.py [hunk 12]
```diff
@@ -121,12 +133,20 @@ def cm2m(cm):
 
 """##### Area ####################################"""
 def sqin2sqm(sqInch):
-    "convert sqare inches to square meters"
-    return sqInch*0.00064516
+    "convert sqaure inches to square meters"
```
Comment (ibell) [inline ACHP/convert_units.py]: typo - square

File: ACHP/convert_units.py [hunk 13]
```diff
@@ -121,12 +133,20 @@ def cm2m(cm):
 
 """##### Area ####################################"""
 def sqin2sqm(sqInch):
-    "convert sqare inches to square meters"
-    return sqInch*0.00064516
+    "convert sqaure inches to square meters"
+    return sqInch*0.00064516    
 def sqm2sqin(sqM):
-    "convert sqare inches to square meters"
+    "convert sqaure inches to square meters"
```
Comment (ibell) [inline ACHP/convert_units.py]: typo -square

**Pull request merged**